### DISCUSSION TITLE
Set velocity to create double jump

### DISCUSSION THREAD

**Discussion by Distractic** [Minestom/Minestom#1581] [Help] [answered]
Hello

I'm checking to create a double jump system.
Basically, the player is allowing to fly, when he tries to fly, we're defining the field `isFlying`, `isAllowFlying` to **false** and apply new velocity. This behavior is called when the event **PlayerStartFlyingEvent** is received.

```kt
class PlayerStartFlyingListener : EventListener<PlayerStartFlyingEvent> {

    override fun eventType(): Class<PlayerStartFlyingEvent> {
        return PlayerStartFlyingEvent::class.java
    }

    override fun run(event: PlayerStartFlyingEvent): EventListener.Result {
        event.player.apply {
            isFlying = false
            isAllowFlying = false
            scheduleNextTick { it.velocity = it.position.direction().mul(3.0).withY(1.5) }
        }
        return EventListener.Result.SUCCESS
    }
}
```

However, the player is sent to the ground instead of powered to the air with the velocity
I checked #1106 and #695 but I didn't find the solution

Have you some tips to do that ?

**Comment by GreatWyrm** [comment]
My guess is that the velocity you are sending to the player is too high, so it's wrapping back around and becoming negative because shorts are used in velocity packets.

I would try something like:
`it.velocity = it.position.direction().mul(0.8).withY(1.5)`.

**Comment by Distractic** [comment, marked as answer]
![double_jump](https://user-images.githubusercontent.com/46402441/208164987-5ab9ba1e-05d6-47a7-ae54-6bd50e294b6a.gif)
That's not the solution .. you can see in the gif, the player is not sent as expected
But in Spigot, the player is correctly sent to the air

**Reply by GreatWyrm** [reply]
...it appears to be the solution? The player appears to be double jumping properly, if you want them to be sent further, you can increase the multiplier until it starts behaving oddly.

As for why it works in Spigot, but is odd in Minestom, I would recommend investigating how Spigot's setVelocity method works.

**Reply by Distractic** [reply]
Whaa  that's true, will a multiply of 200, the expected behavior appears !
Thanks !

**Reply by Distractic** [reply]
```kt
override fun run(event: PlayerStartFlyingEvent): EventListener.Result {
        event.player.apply {
            isFlying = false
            scheduleNextTick { it.velocity = it.position.direction().mul(20.0).withY(20.0) }
        }
        return EventListener.Result.SUCCESS
    }
```
